### PR TITLE
[WIP] Improved and fixed build using MinGW on Windows.

### PR DESCRIPTION
**Pull request by remyroy**
Many elements of the MinGW build on Windows is broken. That might be partially due to upgrades to MinGW and packages updates. This PR attempt to fix and improved the MinGW build process and the related instructions.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Ah. I've had perpetual issues trying to get both MinGW and Code::Blocks working according to instructions, yeah.

**Comment by remyroy** [comment]
The most reliable Windows build is the cross-compiling one from Linux using MXE. It's the one used by Jerkins to build the official experimental. Next is MSYS2. MinGW is pretty low on that list.

**Comment by chaosvolt** [comment]
As I figured. I'm going to assume that cross-compiling from Linux as a Windows user with no Linux experience would be even more of a clusterfuck though. ._.

**Comment by remyroy** [comment]
MSYS2 is pretty straightforward. It's even easier with #13995. It supports sound too now with that PR. Check it out if you prefer a Windows environment.

**Comment by chaosvolt** [comment]
Hmm. I'll have to see if that'll work.

**Comment by remyroy** [comment]
If you need help with the build, come hang out [in the IRC channel](http://webchat.freenode.net/?channels=#CataclysmDDA). I'm often there and I could help you out on the spot.

File: Makefile [hunk 1]
```diff
@@ -373,7 +379,11 @@ else
     CXXFLAGS += $(shell ncursesw5-config --cflags)
     LDFLAGS += $(shell ncursesw5-config --libs)
   else
-    LDFLAGS += -lncurses
+    ifeq ($(NATIVE), win32) # MinGW
+      LDFLAGS += -lncursesw -lpsapi
```
Comment (kevingranade) [inline Makefile]: Why is this new dependency necessary now?

**Comment by remyroy** [comment]
MinGW does not provide any ncurses package. Thomas E. Dickey is providing one on http://invisible-island.net/ncurses/ which depends on psapi as well.

**Comment by remyroy** [comment]
MinGW do provide a pdcurses package which might be an alternative. I'll see if we could use that one instead.

**Comment by kevingranade** [comment]
You say it doesn't provide one, but why was it working previously and not now?

**Comment by chaosvolt** [comment]
If I ever got a compile to work right, I would be able to weigh in on this. <.<

**Comment by remyroy** [comment]
I wish I knew why it was working previously and not now. I'm not quite familiar with CDDA's history but I can come up with a few possible theories:

* It previously worked with instructions not included in `COMPILING.md`.
* It previously worked because MinGW had a ncurses package and they removed it.
* It never worked with ncurses and only with tiles.

**Comment by narc0tiq** [comment]
Perhaps it worked because it was invoking the catacurse emulation, which outputs to a Windows window via GDI+ (I think)? I assume that's what's been changing in the code.

Having a curses-capable terminal on Windows is nontrivial to begin with; it's only "recently" (in the last year or two?) that one or both of the Windows compatibility layers started shipping their own terminal emulator.

**Comment by kevingranade** [comment]
Oh of course, we've been carting around our own ncurses emulation the whole time.
Since catacurses works right now, I'm not particularly eager to add several dependencies, some of which have to be manually installed just to get the same functionality.  Especially especially if that means people need the absolute newest versions of their platform in order to build.

**Comment by remyroy** [comment]
If someone wants to take the lead of this fix for MinGW build on Windows, please do. If not, I'm going to end this and close this PR.

**Comment by remyroy** [comment]
I don't know what can be done here. I guess I will tell people to avoid MingWG and use MSYS2 to build on Windows.

**Comment by karlnp** [comment]
Windows builds don't use ncurses at all - they use catacurses/GDI+, as narcotiq mentioned. This is one of the reasons that MSYS2 is much easier to get working.

**Pull request closed without merging**